cogent/src/Cogent/DataLayout/CodeGen.hs

```diff
@@ -40,7 +59,55 @@ genBoxedGetField
      --
      -- Use this 'CExpr' as the first argument of 'CEFnCall' when you want
      -- to call this getter function.
-genBoxedGetField = undefined
+genBoxedGetField cogentType fieldName = do
+  boxedRecordGetter <- use (boxedRecordGetters . at (cogentType, fieldName))
+  case boxedRecordGetter of
+    Just getFieldFunction -> return getFieldFunction
+    Nothing -> do
+      let TRecord fieldTypes (Boxed _ (RecordLayout fieldLayouts))
+                          = cogentType
+          fieldType       = fst $ (fromList fieldTypes) ! fieldName
+          fieldLayout     = alignLayout $ fst $ fieldLayouts ! fieldName
+      boxCType            <- genType cogentType
+      getFieldFunction    <- genBoxedGetter boxCType fieldType fieldLayout
+      (boxedRecordGetters . at (cogentType, fieldName)) ?= getFieldFunction
+      return getFieldFunction
+
```

Comment:
@zilinc For my own record: Lazy generation of getters. What're the implications for proof automation?

Reply:
Not sure. Will find out when Christine starts figuring out how to do the proofs.